src/renderer/components/NodeDocumentation/NodeDocs.tsx

```diff
@@ -30,61 +30,98 @@ import { NodeExample } from './NodeExample';
 import { SchemaLink } from './SchemaLink';
 
 interface InputOutputItemProps {
-    label: string;
-    description?: string | null;
+    item: Input | Output;
     type: Type;
-    optional: boolean;
-    hasHandle?: boolean;
 }
-const InputOutputItem = memo(
-    ({ label, description, type, optional, hasHandle = true }: InputOutputItemProps) => {
-        if (optional) {
-            // eslint-disable-next-line no-param-reassign
-            type = withoutNull(type);
-        }
 
-        const handleColors = getTypeAccentColors(type);
+const InputOutputItem = memo(({ type, item }: InputOutputItemProps) => {
+    if ('optional' in item && item.optional) {
+        // eslint-disable-next-line no-param-reassign
+        type = withoutNull(type);
+    }
 
-        return (
-            <ListItem my={2}>
-                <HStack>
+    const handleColors = getTypeAccentColors(type);
+
+    const isFileInput = 'kind' in item && item.kind === 'file';
+    const supportedFileTypes = isFileInput ? item.filetypes : [];
+    const isPrimaryInput = isFileInput && 'primaryInput' in item && item.primaryInput;
+
+    return (
+        <ListItem my={2}>
+            <HStack>
+                <Text
+                    fontWeight="bold"
+                    userSelect="text"
+                >
+                    {item.label}
+                </Text>
+                {'optional' in item && item.optional && (
```

Comment:
Not a fan of this code duplication (see without null code). Could make an `optional` variable for it?